netty/src/main/java/io/grpc/netty/AbstractNettyHandler.java

```diff
@@ -108,4 +113,95 @@ private void sendInitialConnectionWindow() throws Http2Exception {
       ctx.flush();
     }
   }
+
+  /**
+   * Class for handling flow control pinging and flow control window updates as necessary
+   */
+  protected class FlowControlPinger {
+
+    private static final int BDP_MEASUREMENT_PING = 1234;
+    private static final int MAX_WINDOW_SIZE = 8 * 1024 * 1024;
+
+    private int pingCount;
+    private int pingReturn;
+    private boolean pinging;
+    private int dataSizeSincePing;
+    private ByteBuf payloadBuff;
```

Comment:
This ByteBuf is never freed, which will cause a memory leak. Since the ping id is constant, you might use [a constant](https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java#L79).